src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -568,19 +568,39 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
         (place, span): (&Place<'tcx>, Span),
         assigned_span: Span,
     ) {
+        let is_arg = if let Place::Local(local) = place {
+            if let LocalKind::Arg = self.mir.local_kind(*local) {
+                true
+            } else {
+                false
+            }
+        } else {
+            false
+        };
```

Comment:
So
much
vertical
space.

Now every time I see stuff like this I'm rewording it with [`is` expressions](https://github.com/rust-lang/rfcs/pull/2260) in my head.
```kotlin
let is_arg = place is Place::Local(local) && self.mir.local_kind(*local) is LocalKind::Arg;
```

Reply:
@petrochenkov To be fair this is just:
```rust
let is_arg = match place {
    Place::Local(local) => self.mir.local_kind(*local) == LocalKind::Arg,
    _ => false
};
```
I think `if let` is overused and match is better in many cases.

Reply:
True, I usually revert to using match in cases like this. Somehow it escaped me here ^_^